packages/holocron/src/updateModuleRegistry.js

```diff
@@ -35,32 +35,62 @@ function defaultGetModulesToUpdate(curr, next) {
 }
 
 export default async function updateModuleRegistry({
-  moduleMap: newModuleMap,
+  moduleMap: unsanitizedModuleMap,
   onModuleLoad = () => null,
   batchModulesToUpdate = (x) => [x],
   getModulesToUpdate = defaultGetModulesToUpdate,
 }) {
   const currentModuleMap = getModuleMap().toJS();
   const modulesToUpdate = batchModulesToUpdate(
-    getModulesToUpdate(currentModuleMap.modules || {}, newModuleMap.modules)
+    getModulesToUpdate(currentModuleMap.modules || {}, unsanitizedModuleMap.modules)
   );
   const flatModulesToUpdate = modulesToUpdate.reduce((acc, batch) => [...acc, ...batch], []);
-
-  let updatedModules = await modulesToUpdate.reduce(async (acc, moduleBatch) => {
-    const loadedModules = await acc;
-    const nextModules = await Promise.all(moduleBatch.map(
-      async (moduleName) => addHigherOrderComponent(
-        await loadModule(moduleName, newModuleMap.modules[moduleName], onModuleLoad)
-      )
+  const problemModules = [];
+  let successfullyLoadedModules = await modulesToUpdate.reduce(async (acc, moduleBatch) => {
+    const previouslyResolvedModules = await acc;
+    const newlyResolvedModules = await Promise.allSettled(moduleBatch.map(
+      async (moduleName) => {
+        try {
+          const loadedModule = await loadModule(
+            moduleName,
+            unsanitizedModuleMap.modules[moduleName],
+            onModuleLoad
+          );
+          return addHigherOrderComponent(loadedModule);
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error(e);
```

Comment:
Add some details/context to this error for the users

Reply:
I don't want to bloat this PR but we should consider adding URLs to our error messages